src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1483,6 +1482,7 @@ public ChangeLogParser createChangeLogParser() {
         private boolean useExistingAccountWithSameEmail;
 //        private GitClientType defaultClientType = GitClientType.GITCLI;
         private boolean showEntireCommitSummaryInChanges;
+        private boolean enablePerformanceImprovement = true; // By default, performance is enabled in git plugin
```

Comment:
This branch was originally created to add performance improvement global switch. This switch has been used to add "opt out" feature for retaining second fetch during the checkout step.